### PR TITLE
Watch later

### PR DESCRIPTION
**Pull request by Shivani-1524**
files to review:
WatchLaterPage
watchlater-utils
watchlater-reducer
watchlater-provider

### CODE REVIEW

File: src/Components/VideoCard/VideoCard.js [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+import { Link, useNavigate } from 'react-router-dom'
+import { VideoDrawer } from '../VideoDrawer/VideoDrawer'
+import './videocard.css'
+import { deleteHistoryVideo } from '../../Utils/history-utils'
+import { addLikedVideo, removeLikedVideo } from '../../Utils/likevideo-utils'
+import { addToWatchlater, deleteFromWatchlater } from '../../Utils/watchlater-utils'
+import { useHistory } from '../../Context/history-provider'
+import { useAuth } from '../../Context/auth-provider'
+import { useLikedVideo } from '../../Context/likevideo-provider'
+import { useWatchlater } from '../../Context/watchlater-provider'
```
Comment (yaswanthmyneni) [inline src/Components/VideoCard/VideoCard.js:11]: suggestion: export all common folder files through index.js

File: src/Components/VideoCard/VideoCard.js [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+import { findElementInData } from '../../Utils/common-utils'
```
Comment (sri-rishi) [inline src/Components/VideoCard/VideoCard.js:12]: Suggestion: you can create index.js file in context and any other folder and can import it from there

File: src/Components/VideoCard/VideoCard.js [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+    const handleHistoryDelete = async (videoId) => {
+        const { data, errorData } = await deleteHistoryVideo(videoId)
+        !errorData[0] ? historyDispatch({ type: "UPDATE_HISTORY", payload: data }) : console.error(errorData[1])
+    }
+    const handleLikeDelete = async (videoId) => {
+        const { data, errorData } = await removeLikedVideo(videoId)
+        !errorData[0] ? likedVideoDispatch({ type: "UPDATE_LIKEDLIST", payload: data?.likes }) : console.error(errorData[1])
+    }
+    const handleWatchlaterDelete = async (videoId) => {
```
Comment (sri-rishi) [inline src/Components/VideoCard/VideoCard.js:32]: Suggestion: you can make separate files for these functions

File: src/Components/VideoCard/videocard.css [hunk 4]
```diff
@@ -0,0 +1,52 @@
+.card{
+    width: min-content;
+    grid-template-rows: auto auto;
+}
+.card .img-container{
+    width: 320px;
+    height: 150px;
+}
```
Comment (sri-rishi) [inline src/Components/VideoCard/videocard.css:9]: Suggestion: can use rem here

**Review by Sahil25061999 — APPROVED** [review]
Code looks good, everything is working properly nothing to suggested, can be merged. Just the same ui suggestion if possible change the color for the remove icons on the cards for better contrast.

**Review by sri-rishi — APPROVED** [review]
Code and UI looks good , feel free to merge

File: src/Components/VideoDrawer/videodrawer.css [hunk 5]
```diff
@@ -0,0 +1,27 @@
+
+  .video-drawer{
+    position: absolute;
+    z-index: 1;
```
Comment (yaswanthmyneni) [inline src/Components/VideoDrawer/videodrawer.css:4]: changes: use css variables for z-indexs

**Review by yaswanthmyneni — APPROVED** [review]
you can merge the pr after changes.

**Comment by Shivani-1524** [comment]
> Code looks good, everything is working properly nothing to suggested, can be merged. Just the same ui suggestion if possible change the color for the remove icons on the cards for better contrast.

thanks and will implement that in the next pr

File: src/Components/VideoCard/VideoCard.js (hunk 2 above)
Comment (Shivani-1524) [reply src/Components/VideoCard/VideoCard.js:12]: done

**Pull request merged**